digitalocean/app/resource_app.go

```diff
@@ -314,3 +347,223 @@ func waitForAppDeployment(client *godo.Client, id string, timeout time.Duration)
 
 	return fmt.Errorf("timeout waiting for app (%s) deployment", id)
 }
+
+func mapAppLevelAlerts(d *schema.ResourceData) []*godo.AppAlert {
+	spec, ok := d.GetOk("spec")
+	if !ok {
+		return nil
+	}
+
+	specList := spec.([]interface{})
+	if len(specList) == 0 {
+		return nil
+	}
+	specMap, ok := specList[0].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	alert, ok := specMap["alert"]
+	if !ok {
+		return nil
+	}
+
+	alertList, ok := alert.([]interface{})
+	if !ok {
+		return nil
+	}
+
+	var alertDetails []*godo.AppAlert
+
+	for _, alertItem := range alertList {
+		alertData, ok := alertItem.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		alertDetail := &godo.AppAlert{
+			Spec:          &godo.AppAlertSpec{},
+			Emails:        []string{},
+			SlackWebhooks: []*godo.AppAlertSlackWebhook{},
+		}
+
+		for key, value := range alertData {
+			switch key {
+			case "rule":
+				if rule, ok := value.(string); ok {
+					alertDetail.Spec.Rule = godo.AppAlertSpecRule(rule)
+				}
+			case "notifications":
+				if notifications, ok := value.([]interface{}); ok {
+					emails, slackWebhooks := extractEmailAndSlackNotifications(notifications)
+
+					if len(emails) == 0 && len(slackWebhooks) == 0 {
+						continue
+					}
+
+					alertDetail.Emails = emails
+					alertDetail.SlackWebhooks = slackWebhooks
+				}
+			}
+		}
+
+		alertDetails = append(alertDetails, alertDetail)
+	}
+
+	return alertDetails
+}
+
+func syncAppAlertDestinations(d *schema.ResourceData, client *godo.Client, appID string) error {
+	appConfigurationAlert := mapAppLevelAlerts(d)
+	computeComponentAlerts := mapComputeComponentAlert(d)
+
+	hasAppLevelAlerts := len(appConfigurationAlert) > 0 && (len(appConfigurationAlert[0].Emails) > 0 || len(appConfigurationAlert[0].SlackWebhooks) > 0)
+	hasComputeComponentAlerts := len(computeComponentAlerts) > 0 && (len(computeComponentAlerts[0].Emails) > 0 || len(computeComponentAlerts[0].SlackWebhooks) > 0)
+
+	var alerts []*godo.AppAlert
+	if hasAppLevelAlerts || hasComputeComponentAlerts {
+		appAlerts, _, err := client.Apps.ListAlerts(context.Background(), appID)
+		if err != nil {
+			return err
+		}
+		alerts = appAlerts
+	}
+
+	if hasAppLevelAlerts {
+		err := resourceDigitalOceanAlertDestinationUpdate(appID, alerts, appConfigurationAlert, client, func(a, b *godo.AppAlert) bool {
+			return string(a.Spec.Rule) == string(b.Spec.Rule)
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	if hasComputeComponentAlerts {
+		err := resourceDigitalOceanAlertDestinationUpdate(appID, alerts, computeComponentAlerts, client, func(a, b *godo.AppAlert) bool {
+			return string(a.Spec.Rule) == string(b.Spec.Rule)
+		})
```

Comment:
I'm a bit unsettled on this part. It is only checking the rule. However, an user may set multiple alerts for the same rule different values.

I could've checked values and other parameters, but then if someone changes those as well, then this match will fail. 
Open to alternatives.

Reply:
One idea was to make compare all 3 values and document it. If they are updating alert destinations (only destinations), the user must make sure of this. 
Worst case the user would need to run one more deployment.